### PR TITLE
added agg_source_urls command

### PR DESCRIPTION
**Pull request by srossross**
alternative to #55

### CODE REVIEW

File: score/cli.py [hunk 1]
```diff
@@ -153,5 +154,34 @@ def conda(num_partitions, partition, output, channel):
     click.echo("Scraping completed.")
 
 
+@cli.command()
+@click.option(
+    "-o",
+    "--output",
+    default=os.path.join(OUTPUT_ROOT, "github-urls.parquet"),
```
Comment (srossross) [inline score/cli.py:161]: The output of an aggregation should be a single file. partitions here do not make sense

File: score/cli.py [hunk 2]
```diff
@@ -153,5 +154,34 @@ def conda(num_partitions, partition, output, channel):
...
+def agg_source_urls(input, output):
+    click.echo("Aggregating data")
+
+    db = duckdb.connect()
+    # Public dataset / empty secret
+    db.execute("CREATE SECRET (TYPE GCS);")
+
+    df = db.query(
+        f"""
+    select distinct source_url from read_parquet('{input}/pypi-json/*/*.parquet');
```
Comment (srossross) [inline score/cli.py:179]: Will need to add `/conda` parquet files 

if you use pypi partions how would be join conda and npm source files?

File: score/data_retrieval/json_scraper.py [hunk 3]
```diff
@@ -51,28 +54,41 @@ def get_package_data(package_name):
         "requires_python": info.get("requires_python", None),
         "version": info.get("version", None),
         "yanked_reason": info.get("yanked_reason", None),
-        "source_url": extract_github_repo(info.get("project_urls", {})),
+        "source_url": source_url,
+        "source_url_key": source_url_key,
```
Comment (srossross) [inline score/data_retrieval/json_scraper.py:58]: metadata to debug

File: score/data_retrieval/json_scraper.py [hunk 4]
```diff
@@ -51,28 +54,41 @@ def get_package_data(package_name):
...
-        return None
-    for url in project_urls.values():
-        if url and GITHUB_REPO_PATTERN.match(url):
-            return url
-    return None
+        return None, None
+
+    project_urls = {k.lower(): v for k, v in project_urls.items()}
+
+    for key in ["code", "repository", "source", "homepage"]:
```
Comment (srossross) [inline score/data_retrieval/json_scraper.py:84]: github 'issue tracker' does mean the source is kept there. lets stick to some standard keys. If we find there are other keys we can add them later when someone notices their package is not linked to the source properly

**Review by karamba228 — APPROVED** [review]

**Pull request merged**